streaming/src/main/scala/org/apache/spark/streaming/receiver/ReceiverSupervisor.scala

```diff
@@ -161,22 +163,45 @@ private[streaming] abstract class ReceiverSupervisor(
     }
   }
 
-  /** Check if receiver has been marked for stopping */
+  /** Check if receiver has been marked for stopping. */
   def isReceiverStarted() = {
     logDebug("state = " + receiverState)
     receiverState == Started
   }
 
-  /** Check if receiver has been marked for stopping */
+  /** Check if receiver has been marked for stopping. */
   def isReceiverStopped() = {
     logDebug("state = " + receiverState)
     receiverState == Stopped
   }
 
+  /** Due to SPARK-5205, we need `TaskContext` to check if the task is interrupted. */
+  def setTaskContext(context: TaskContext) = {
```

Comment:
Instead of adding a setter method, why not pass the TaskContext into the ReceiverSupervistor's constructor?